mtags/src/main/scala-2/scala/meta/internal/pc/PcReferencesProvider.scala

```diff
@@ -0,0 +1,57 @@
+package scala.meta.internal.pc
+
+import scala.meta.internal.jdk.CollectionConverters._
+import scala.meta.internal.mtags.MtagsEnrichments._
+import scala.meta.pc.DefinitionResult
+import scala.meta.pc.OffsetParams
+import scala.meta.pc.VirtualFileParams
+
+import org.eclipse.{lsp4j => l}
+
+class PcReferencesProvider(
+    compiler: MetalsGlobal,
+    params: OffsetParams,
+    targetFiles: List[VirtualFileParams],
```

Comment:
I wonder if we shouldn't read the files on demand instead of sending them in the params here. There are cases when we will not need targetFiles to be read at all. So maybe we should just read it here. Also sending too many large files might cause OOM maybe?

Reply:
Yeah, I think it's reasonable to ask for batches instead of giving all at once.